flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/JarRunHandler.java

```diff
@@ -138,12 +154,22 @@ public JarRunHandler(
 			});
 	}
 
-	private static SavepointRestoreSettings getSavepointRestoreSettings(
-			final @Nonnull HandlerRequest<EmptyRequestBody, JarRunMessageParameters> request)
+	private SavepointRestoreSettings getSavepointRestoreSettings(
+			final @Nonnull HandlerRequest<JarRunRequestBody, JarRunMessageParameters> request)
 				throws RestHandlerException {
 
-		final boolean allowNonRestoredState = getQueryParameter(request, AllowNonRestoredStateQueryParameter.class, false);
-		final String savepointPath = getQueryParameter(request, SavepointPathQueryParameter.class);
+		final JarRunRequestBody requestBody = request.getRequestBody();
+
+		final boolean allowNonRestoredState = fromRequestBodyOrQueryParameter(
+			requestBody.getAllowNonRestoredState(),
+			() -> getQueryParameter(request, AllowNonRestoredStateQueryParameter.class),
+			false,
+			log);
+		final String savepointPath = fromRequestBodyOrQueryParameter(
```

Comment:
This could result in unexpected NullPointerExceptions when retrieving a primitive, like in the following example:
```
fromRequestBodyOrQueryParameter(
		requestBody.getParallelism(),
		() -> getQueryParameter(request, ParallelismQueryParameter.class)
		log);
```
The explicit default argument prevents that from happening.

Reply:
How does this prevent the scenario i described?

Reply:
True, making the specification of the default value explicit will better guard against this. In the other case, the user would have to know the difference between these methods and which to apply to primitive and non-primitive values. Alright, then it's good to go from my side.